IceCream/Classes/SyncEngine.swift

```diff
@@ -79,8 +79,8 @@ extension SyncEngine {
     
     /// Push all existing local data to CloudKit
     /// You should NOT to call this method too frequently
-    public func pushAll() {
-        databaseManager.syncObjects.forEach { $0.pushLocalObjectsToCloudKit() }
+    public func pushAll(completionHandler: ((Error?) -> Void)? = nil) {
```

Comment:
it seems like this completion handler will get called many times because it's in the forEach loop. Would we want to add some sort of control flow so it only get's called once after all syncObjects are pushed?

Reply:
@scottsykora ah yes, that makes sense. i was thinking of trying to execute each push and if they complete with return errors, collect them all and put them in a custom error.  does that sound reasonable?

Reply:
that sounds reasonable to me

Reply:
@premedrchang yeah, collecting all the errors makes sense, or just passing back the first error once they're all done would also work. Whichever is the better balance of practicality and usefulness in the errors.

Reply:
@jacobbullock @scottsykora soo... i implemented the same functionality as pushAll() in our PixiteMetaData pod's pushLocalRecordsToCloud() but now that handles the multiple SyncObject pushes and collects all the errors that come back.  the bonus is our method is async. 

we don't actually use IceCream's pushAll() method anywhere and i only messed with it after adding a completion block to the SyncObject's push method.  

SOOO, i'm actually thinking of reverting this back to what it originally was.

Reply:
Cool, I'm good with that.